python/paddle/tensor/manipulation.py

```diff
@@ -6749,3 +6749,108 @@ def select_scatter(x, values, axis, index, name=None):
         )
 
         return output
+
+
+def slice_scatter(x, value, axis=0, start=None, stop=None, step=1, name=None):
+    """
+    Embeds the value tensor into x at the given axis. Returns a new tensor instead of a view.
+
+    Args:
+        x (Tensor) : The input Tensor. Supported data types are `bool`, `float16`, `float32`, `float64`, `uint8`, `int8`, `int16`, `int32`, `int64`, `bfloat16`, `complex64`, `complex128`.
+        value (Tensor) : The tensor to embed into x. Supported data types are `bool`, `float16`, `float32`, `float64`, `uint8`, `int8`, `int16`, `int32`, `int64`, `bfloat16`, `complex64`, `complex128`.
+        axis (int) : the dimension to insert the value. Default is 0.
+        start (int, optional) : the start index of where to insert. Default is `None` which will be converted to `0`.
+        stop (int, optional) : the stop index of where to insert. Default is `None` which will be converted to `x.shape[axis]`.
+        step (int, optional) : the step for each insert. Default is 1.
+        name (str, optional): Name for the operation (optional, default is None).
+
+    Returns:
+        Tensor, same dtype and shape with x
+
+    Examples:
+        .. code-block:: python
+
+            >>> import paddle
+
+            >>> x = paddle.zeros((8, 8))
+            >>> value = paddle.ones((8, 2))
+            >>> res = paddle.slice_scatter(x, value, axis=1, start=2, stop=6, step=2)
+            >>> print(res)
+            Tensor(shape=[8, 8], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[0., 0., 1., 0., 1., 0., 0., 0.],
+             [0., 0., 1., 0., 1., 0., 0., 0.],
+             [0., 0., 1., 0., 1., 0., 0., 0.],
+             [0., 0., 1., 0., 1., 0., 0., 0.],
+             [0., 0., 1., 0., 1., 0., 0., 0.],
+             [0., 0., 1., 0., 1., 0., 0., 0.],
+             [0., 0., 1., 0., 1., 0., 0., 0.],
+             [0., 0., 1., 0., 1., 0., 0., 0.]])
+
+    """
+    if x.ndim != value.ndim:
+        raise ValueError(
+            f"The input x and value should have save dimension, but got input of {x.ndim} and value of {value.ndim}."
+        )
+
+    x_shape = x.shape
+    value_shape = value.shape
+
+    start = 0 if start is None else start
+    stop = x_shape[axis] if stop is None else stop
+
+    index = list(range(start, stop, step))
+    exp_shape = [*x_shape[:axis], len(index), *x_shape[axis + 1 :]]
+    if tuple(exp_shape) != tuple(value_shape):
+        raise ValueError(
+            "The value.shape should be same of [*x_shape[:axis], len(index), *x_shape[axis+1:]],"
+            f"but got value.shape of {value.shape} and slice shape {exp_shape}."
+        )
```

Comment:
we should support if the shape of `value` can broadcast to the shape of `exp_shape`, and this check will be completed in `set_value`, so there is no need to add this check here, which actually limits the functionality.